rdtools/test/irradiance_rescale_test.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+from pandas.testing import assert_series_equal
+from rdtools import irradiance_rescale
+from rdtools.normalization import ConvergenceError
+import pytest
+
+
+@pytest.fixture
+def simple_irradiance():
+    times = pd.date_range('2019-06-01 12:00', freq='15T', periods=5)
+    time_series = pd.Series([1,2,3,4,5], index=times)
+    return time_series
+
+
+@pytest.mark.parametrize("method", ['iterative', 'single_opt'])
+def test_rescale(method, simple_irradiance):
+    # test basic functionality
+    modeled = simple_irradiance
+    measured = 1.05 * simple_irradiance
+    rescaled = irradiance_rescale(measured, modeled, method=method)
+    expected = measured
+    assert_series_equal(rescaled, expected, check_exact=False)
+    
+
+def test_warn(simple_irradiance):
+    with pytest.warns(UserWarning,
+                      match="underlying calculations for irradiance_rescale"):
+        _ = irradiance_rescale(simple_irradiance, simple_irradiance)
+
+
+def test_max_iterations(simple_irradiance):
```

Comment:
Is this test redundant with `test_non_convergence()`?

Reply:
I suppose the original intent is redundant since both of them verify that the `max_iterations` parameter changes function behavior, and `test_non_convergence()` is the better test.  

This makes me wonder if the specific behavior `test_max_iterations()` is testing should be changed.   One could imagine a rescale function that returns the input unchanged (or raises ConvergenceError) when `max_iterations=0` and it seems the reason this function doesn't do that is just an implementation detail rather than intended design.  Should we change it?  If not, I think we should document that max_iterations must be greater than zero.

Reply:
Thanks for pointing this out. Let's take this opportunity to think through rescale and perhaps align its behavior more closely with `scipy.optimize.minimize` for example. I suggest we:

- allow it to work with `max_iterations=0`
- Raise a warning rather than an error when convergence isn't reached within `max_iterations`
-expose `convergence_threshold` as an optional parameter

What do you think of these suggestions?

Reply:
+1 to 1 and 3.  For changing the error to a warning, I could go either way.  I'm pretty desensitized to numerical warnings and would likely not pay attention to something similar from rdtools -- the way I use python, it's unusual that a warning is actually about something I care about.  Usually it's about divide-by-zero at night or something that is fine to ignore.

IMO issuing a warning but returning the rescaled data anyway is some form of endorsement that the results are still useful, but I'd expect non-convergence to be associated with something funky with the data that ought to be fixed before analyzing.  With `max_iterations` and `convergence_threshold` exposed, do you think that returning non-converged rescalings is useful?

Reply:
I could also be convinced either way on 2. And it is more conservative to raise an error, so let's stick with the error on that one.